File: 1047_Remove All Adjacent Duplicates In String/RemoveAllAdjacentDuplicatesInString.cs
public class Solution {
    public string RemoveDuplicates(string s) {
        if(string.IsNullOrEmpty(s)) return s;
        var stack = new Stack<char>();
        foreach(var c in s)
        {
            if(stack.Count == 0){
                stack.Push(c);
            }else{
                if(stack.Peek() == c){
                    stack.Pop();
                }else{
                    stack.Push(c);
                }
            }
        }
        
        var chars = new char[stack.Count];
        for(int i=chars.Length-1;i>=0;i--)
            chars[i] = stack.Pop();
        
        return new String(chars);
    }
}

File: 1047_Remove All Adjacent Duplicates In String/RemoveAllAdjacentDuplicatesInString.java
class Solution {
    public String removeDuplicates(String s) {
        if(s == null || s.length() <= 1) return s;
        Stack<Character> stack = new Stack<Character>();
        
        for(char c : s.toCharArray()) {
            if(stack.size() > 0 && stack.peek() == c) {
                stack.pop();
            }else {
                stack.push(c);
            }
        }
        
        StringBuilder sb = new StringBuilder();
        while(stack.size() > 0)
            sb.insert(0, stack.pop());
        
        return sb.toString();
    }
}